core/src/main/java/org/apache/calcite/rel/metadata/RelMdColumnUniqueness.java

```diff
@@ -378,12 +402,36 @@ public Boolean areColumnsUnique(Intersect rel, RelMetadataQuery mq,
     return null;
   }
 
+  /**
+   * If an input's unique column value is returned (passed through) by an aggregation function, then
+   * the result of the function is also unique.
+   *
+   * @param columns         the columns to test for uniqueness
+   * @param rel             the Aggregate {@link RelNode}
+   * @param inputUniqueKeys the set of unique key sets of the Aggregate input (child RelNode)
+   *
+   * @return whether the input columns are unique by containing a unique aggregated column
+   */
+  private boolean columnsContainUniqueAgg(ImmutableBitSet columns, Aggregate rel,
+      Set<ImmutableBitSet> inputUniqueKeys) {
+    List<AggregateCall> aggCallList = rel.getAggCallList();
+    for (int aggIndex = 0; aggIndex < aggCallList.size(); aggIndex++) {
```

Comment:
This `for` loop seems backwards to me. It would make more sense to iterate over `columns`, and ask whether each is a pass-through aggregate function. Whereas you loop over all aggregate functions.

Reply:
When you have pushed a new commit, please make a comment on the Jira case and I will re-review. This PR seems close to completion.